awscli/customizations/s3/filegenerator.py

```diff
@@ -320,7 +323,11 @@ def _list_single_object(self, s3_path):
         # instead use a HeadObject request.
         bucket, key = find_bucket_key(s3_path)
         try:
-            response = self._client.head_object(Bucket=bucket, Key=key)
+            params = {'Bucket': bucket, 'Key': key}
+            if self.sse_c:
```

Comment:
Well that's unfortunate.  Seems like we're missing a level of abstraction here.  Maybe it's worth abstracting these away as a parameter object so we don't have to change this class in the future if new params are required for head_object with new features.

Either that or I think it might be worth pulling this whole "metadata for a single object" concept out into its own object.  This would isolate future changes even more.